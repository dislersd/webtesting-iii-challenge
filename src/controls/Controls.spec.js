import React from "react";
import ReactDOM from "react-dom";
import "jest-dom/extend-expect";
import "react-testing-library/cleanup-after-each";
import renderer from "react-test-renderer";
import Controls from "./Controls";
import { render, cleanup } from "react-testing-library";
import { fireEvent } from "react-testing-library/dist";

afterEach(cleanup);

describe("<Controls /> Snapshot", () => {
  it("matched snapshot", () => {
    const tree = renderer.create(<Controls />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("<Controls />", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Controls />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it("Renders Sucessfully", () => {
    render(<Controls />);
  });
  it("open-button fires a callback on click", () => {
    const banana = jest.fn();
    const { getByTestId } = render(<Controls toggleClosed={banana}/>);
    fireEvent.click(getByTestId('openButton'))
    expect(banana).toHaveBeenCalled();
    expect(banana).toHaveBeenCalledTimes(1);
  });
  it("open-button fires a callback on click", () => {
    const mock = jest.fn();
    const { getByTestId, debug } = render(<Controls toggleClosed={mock}/>);
    fireEvent.click(getByTestId('openButton'))
    expect(mock).toHaveBeenCalled();
    expect(mock).toHaveBeenCalledTimes(1);
  });
  it("Renders the control panel div", () => {
    const {getByTestId} = render(<Controls />);
    const controlPanel = getByTestId('controlPanel')
    expect(controlPanel).toBeDefined();
  })
});