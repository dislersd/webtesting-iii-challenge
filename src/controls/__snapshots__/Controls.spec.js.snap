// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Controls /> Snapshot matched snapshot 1`] = `
<div
  className="controls panel"
  data-testid="controlPanel"
>
  <button
    className="toggle-btn"
    data-testid="lockButton"
    disabled={true}
  >
    Lock Gate
  </button>
  <button
    className="toggle-btn"
    data-testid="openButton"
  >
    Close Gate
  </button>
</div>
`;
